CMakeLists.txt

```diff
@@ -372,8 +374,10 @@ if(BUILD_FUZZ_BINARY)
   )
 endif()
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "Generic")
```

Comment:
I imagine that expanding this build scenario might enable building `boost` and `libevent` in depends, no?

If so, the Boost check should have another option to be disabled.

Additionally, mind considering this comment: https://github.com/bitcoin/bitcoin/blob/1172bc4157eefe80d1aaf0b56459857ec651e535/cmake/module/AddBoostIfNeeded.cmake#L7-L8

which, in turn, was based on this legacy code:https://github.com/bitcoin/bitcoin/blob/32efe850438ef22e2de39e562af557872a402c31/configure.ac#L1247-L1251